### PR TITLE
Enable usage of machine set CRATES var

### PR DESCRIPTION
**Pull request by tcharding**
Repos using the `run_task` script would like to set the `CRATES` env var by using

```bash
REPO_DIR=$(git rev-parse --show-toplevel)
CRATES="$(cargo metadata --no-deps --format-version 1 | jq -j -r '.packages | map(.manifest_path | rtrimstr("/Cargo.toml") | ltrimstr("'"$REPO_DIR"'/")) | join(" ")')"
```

I don't know exactly why but this results in some sort of square braces data type that seems to only work in a loop using `for crate in $CRATES` instead of the `for crate in ${CRATES[@]}` like we currently have.

### CODE REVIEW

**Comment by apoelstra** [comment]
Using `$PWD` like that will mean that the script doesn't work if it's run from the wrong directory. I think we have an env var set somewhere already which uses `git rev-parse --show-toplevel` or something to orient our scripts around a fixed directory.

**Comment by tcharding** [comment]
I'll come back to this after #11 merges.

**Comment by tcharding** [comment]
The commit hash from this branch is now used in https://github.com/rust-bitcoin/rust-bitcoin/pull/3201 to verify this works using CI.

**Review by apoelstra — APPROVED** [review]
ACK 534e4977007760fafe5154bc31acae08cb535fbb

**Review by apoelstra — APPROVED** [review]
ACK 534e4977007760fafe5154bc31acae08cb535fbb

**Comment by tcharding** [comment]
Shall we give @Kixunil a chance to take a look, especially since I copied his code to get the looping to work?

**Review by Kixunil — APPROVED** [review]
ACK 534e4977007760fafe5154bc31acae08cb535fbb

However I wonder why a global variable is used and why isn't the command to obtain the list of crates ran by this script.

**Comment by apoelstra** [comment]
Also Kix ought to be a maintainer on this repo.

**Pull request merged**
